### PR TITLE
MOBILE-4250 module: Add module summary to preview module page

### PR DESCRIPTION
**Pull request by crazyserver**

### CODE REVIEW

File: src/core/features/course/pages/module-preview/module-preview.ts [hunk 1]
```diff
@@ -92,6 +96,42 @@ export class CoreCourseModulePreviewPage implements OnInit {
...
+        const data = await CoreDomUtils.openSideModal<CoreCourseModuleSummaryResult>({
+            component: CoreCourseModuleSummaryComponent,
+            componentProps: {
+                moduleId: this.module.id,
+                module: this.module,
+                description: this.module.description,
+                component: this.module.modname,
+                courseId: this.courseId,
+                displayOptions: {
+                    // Show description on summary if not shown on the page.
```
Comment (dpalou) [inline src/core/features/course/pages/module-preview/module-preview.ts:116]: I don't understand this comment. You're always saying displayDescription=false, right?

File: src/core/features/course/pages/module-preview/module-preview.ts [hunk 2]
```diff
@@ -92,6 +96,42 @@ export class CoreCourseModulePreviewPage implements OnInit {
...
+                    displayDescription: false,
+                    displayBlog: false,
+                },
+            },
+        });
+
+        if (data) {
+            if (!this.loaded && (data.action == 'refresh' || data.action == 'sync')) {
```
Comment (dpalou) [inline src/core/features/course/pages/module-preview/module-preview.ts:124]:
```suggestion
            if (this.loaded && (data.action == 'refresh' || data.action == 'sync')) {
```

Shouldn't this be the opposite? We want to refresh if data is loaded. Also, can we have a 'sync' action in module preview?

**Review by dpalou — CHANGES_REQUESTED** [review]

File: src/core/features/course/pages/module-preview/module-preview.ts (hunk 1 above)
Comment (crazyserver) [reply src/core/features/course/pages/module-preview/module-preview.ts:116]: Copy paste :-p

File: src/core/features/course/pages/module-preview/module-preview.ts (hunk 2 above)
Comment (crazyserver) [reply src/core/features/course/pages/module-preview/module-preview.ts:124]: true... :-(

**Review by dpalou — APPROVED** [review]

**Pull request merged**